src/Numerics/DGMethods/DGModel.jl

```diff
@@ -117,13 +199,219 @@ Computes the tendency terms compatible with `IncrementODEProblem`
+                        dgfvm.state_gradient_flux,
+                        dependencies = comp_stream,
+                    )
+            end
+        end
+
+        if num_state_gradient_flux > 0
+            # update_aux_diffusive may start asynchronous work on the compute device
+            # and we synchronize those here through a device event.
+            wait(device, comp_stream)
+            update_auxiliary_state_gradient!(
+                dgfvm,
+                dgfvm.balance_law,
+                state_prognostic,
+                t,
+                dgfvm.grid.topology.realelems,
+            )
+            comp_stream = Event(device)
+        end
+    end
+
+    ###################
+    # RHS Computation #
+    ###################
+    comp_stream = launch_volume_tendency!(
+        dgfvm,
+        tendency,
+        state_prognostic,
+        t,
+        α,
+        β;
+        dependencies = (comp_stream,),
+    )
+
+    comp_stream = launch_interface_tendency!(
+        dgfvm,
+        tendency,
+        state_prognostic,
+        t,
+        α,
+        β;
+        surface = :interior,
+        dependencies = (comp_stream,),
+    )
+
+    if communicate
+        if num_state_gradient_flux > 0
+            exchange_state_gradient_flux = MPIStateArrays.end_ghost_exchange!(
+                dgfvm.state_gradient_flux;
+                dependencies = exchange_state_gradient_flux,
+            )
+
+            # update_aux_diffusive may start asynchronous work on the
+            # compute device and we synchronize those here through a device
+            # event.
+            wait(device, exchange_state_gradient_flux)
+            update_auxiliary_state_gradient!(
+                dgfvm,
+                dgfvm.balance_law,
+                state_prognostic,
+                t,
+                dgfvm.grid.topology.ghostelems,
+            )
+            exchange_state_gradient_flux = Event(device)
+        else
+            exchange_state_prognostic = MPIStateArrays.end_ghost_exchange!(
+                state_prognostic;
+                dependencies = exchange_state_prognostic,
+            )
+
+            # update_aux may start asynchronous work on the compute device and
+            # we synchronize those here through a device event.
+            wait(device, exchange_state_prognostic)
+            update_auxiliary_state!(
+                dgfvm,
+                dgfvm.balance_law,
+                state_prognostic,
+                t,
+                dgfvm.grid.topology.ghostelems,
+            )
+            exchange_state_prognostic = Event(device)
+        end
+    end
+
+    comp_stream = launch_interface_tendency!(
+        dgfvm,
+        tendency,
+        state_prognostic,
+        t,
+        α,
+        β;
+        surface = :exterior,
+        dependencies = (
+            comp_stream,
+            exchange_state_prognostic,
+            exchange_state_gradient_flux,
+            #JK exchange_Qhypervisc_grad,
```

Comment:
Is this what needs to be turned on (i.e. decommented) when adding hyperdiffusion back? Should leave a note/comment?

Reply:
Hopefully, this and a few other places.

We also need to add a test for hyperdiffusion with DG-FV.

I don't think that there is a ton that needs to happen since hyperdiffusion is (I believe), for now just going to be on the DG side.

Hopefully will be an easy follow-up PR!

Reply:
Okay, I fixed it up!

Reply:
You can bors r+ at your convenience

Reply:
Thanks.